chunk.go

```diff
@@ -0,0 +1,38 @@
+package functools
+
+import (
+	"math"
+)
+
+// Chunk consumes a slice of a generic type and
+// returns a slice composed of multiple chunks of a user-specified size
+//
+// Vacuously, empty slices return empty regardless of the chunk size.
```

Comment:
Nit: change wording to `Vacuously, empty slices return an empty slice of slices regardless of the chunk size`